src/webgpu/shader/validation/expression/call/builtin/smoothstep.spec.ts

```diff
@@ -0,0 +1,209 @@
+
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      expectedResult,
+      [type.create(t.params.value1), type.create(t.params.value2), type.create(0)],
+      t.params.stage
+    );
+  });
+
+g.test('argument_types')
+  .desc(
+    `
+Validates that scalar and vector arguments are rejected by ${builtin}() if not float type or vecN<float type>
+`
+  )
+  .params(u => u.combine('type', keysOf(kArgumentTypes)))
+  .beforeAllSubcases(t => {
+    if (scalarTypeOf(kArgumentTypes[t.params.type]) === Type.f16) {
+      t.selectDeviceOrSkipTestCase('shader-f16');
+    }
+  })
+  .fn(t => {
+    const type = kArgumentTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ isFloatType(elementTypeOf(type)),
+      [type.create(0), type.create(1), type.create(2)],
+      'constant',
+      /* returnType */ concreteTypeOf(type)
+    );
+  });
+
+const kTests = {
+  valid: {
+    src: `_ = ${builtin}(0.0, 42.0, 0.5);`,
+    pass: true,
+  },
+  alias: {
+    src: `_ = ${builtin}(f32_alias(0), f32_alias(42), f32_alias(0.5));`,
+    pass: true,
+  },
+  bool: {
+    src: `_ = ${builtin}(false, false, false);`,
+    pass: false,
+  },
+  i32: {
+    src: `_ = ${builtin}(1i, 1i, 1i);`,
+    pass: false,
+  },
+  u32: {
+    src: `_ = ${builtin}(1u, 1u, 1u);`,
+    pass: false,
+  },
+  vec_bool: {
+    src: `_ = ${builtin}(vec2<bool>(false, true), vec2<bool>(false, true), vec2<bool>(false, true));`,
+    pass: false,
+  },
+  vec_i32: {
+    src: `_ = ${builtin}(vec2<i32>(1, 1), vec2<i32>(1, 1), vec2<i32>(1, 1));`,
+    pass: false,
+  },
+  vec_u32: {
+    src: `_ = ${builtin}(vec2<u32>(1, 1), vec2<u32>(1, 1), vec2<u32>(1, 1));`,
+    pass: false,
+  },
+  vec_f32: {
+    src: `_ = ${builtin}(vec2<f32>(0, 0), vec2<f32>(1, 1), vec2<f32>(1, 1));`,
+    pass: true,
+  },
+  matrix: {
+    src: `_ = ${builtin}(mat2x2(1, 1, 1, 1), mat2x2(1, 1, 1, 1), mat2x2(1, 1, 1, 1));`,
+    pass: false,
+  },
+  atomic: {
+    src: ` _ = ${builtin}(a, a, a);`,
+    pass: false,
+  },
+  array: {
+    src: `var a: array<bool, 5>;
+            _ = ${builtin}(a, a, a);`,
+    pass: false,
+  },
+  array_runtime: {
+    src: `_ = ${builtin}(k.arry, k.arry, k.arry);`,
+    pass: false,
+  },
+  struct: {
+    src: `var a: A;
+            _ = ${builtin}(a, a, a);`,
+    pass: false,
+  },
+  enumerant: {
+    src: `_ = ${builtin}(read_write, read_write, read_write);`,
+    pass: false,
+  },
+  ptr: {
+    src: `var<function> a = 1.0;
+            let p: ptr<function, f32> = &a;
+            _ = ${builtin}(p, p, p);`,
+    pass: false,
+  },
+  ptr_deref: {
+    src: `var<function> a = 1.0;
+            let p: ptr<function, f32> = &a;
+            _ = ${builtin}(*p, *p, *p);`,
+    pass: true,
+  },
+  sampler: {
+    src: `_ = ${builtin}(s, s, s);`,
+    pass: false,
+  },
+  texture: {
+    src: `_ = ${builtin}(t, t, t);`,
+    pass: false,
+  },
+  no_params: {
```

Comment:
`no_args` The number of parameters never changes.

Reply:
Done.